### PR TITLE
 Rename config-kafka-features feature flags to be consistent with eventing core feature flag names

### LINKED ISSUE #3509: Rename config-kafka-features feature flags to be consistent with eventing core feature flag names
**Problem**

The Kafka feature flag names are not consistent with Eventing core, we're using dots to separate words while eventing core is using a hyphen.

We should rename feature flags in [1] and [2] as follow:

```yaml
  dispatcher-rate-limiter: "disabled"
  dispatcher-ordered-executor-metrics: "disabled"
  controller-autoscaler-keda: "disabled"
  triggers-consumergroup-template: "knative-trigger-{{ .Namespace }}-{{ .Name }}"
  brokers-topic-template: "knative-broker-{{ .Namespace }}-{{ .Name }}"
  channels-topic-template: "knative-messaging-kafka.{{ .Namespace }}.{{ .Name }}"
```

**In the code, we should make sure that we keep parsing the old deprecated feature flag names if the new ones are not present to avoid breaking changes with something like this (for each feature flag):**

```golang
err := configmap.Parse(cm.Data, 
 		asFlag("dispatcher.rate-limiter", &nc.features.DispatcherRateLimiter), 
 		asFlag("dispatcher-rate-limiter", &nc.features.DispatcherRateLimiter), , 
 	)
``` 

[1] https://github.com/knative-extensions/eventing-kafka-broker/blob/main/control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml
[2] https://github.com/knative-extensions/eventing-kafka-broker/blob/7b581b7807df14b27b7100293706da8269c04c45/control-plane/pkg/apis/config/features.go#L79-L90

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
Admin

**Exit Criteria**

Feature flags have different names

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1

**Additional context (optional)**

/good-first-issue

### PR DESCRIPTION
**Pull request by cdalar**
Fixes #3509 

## Proposed Changes

- Rename config-kafka-features feature flags to be consistent with eventing core feature flag names
 
```release-note 
Align config-kafka-features flags to eventing core feature flags.  
```

### CODE REVIEW

**Comment by creydr** [comment]
Hello @cdalar,
thanks for your PR. As we squash the commits and use the PR title as commit message, can you please rename the PR to something meaningful? And as this is a user facing change, can you please add a release note comment in the PR description:

via 
https://github.com/knative-extensions/eventing-kafka-broker/blob/ddd03f9fcdb16b928e5662ff1f3c9ff3ca97f176/.github/pull-request-template.md?plain=1#L34-L36

/ok-to-test

File: control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml
Comment (pierDipi) [inline control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml:1]: The annotation `knative.dev/example-checksum` needs to change its value to `cf3393de`

You can replace the value or run `hack/update-codgen.sh`

**Review by pierDipi — COMMENTED** [review]
Thanks!

**Comment by pierDipi** [comment]
Test failures are unrelated

File: control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml
Comment (cdalar) [reply control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml:1]: updated.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by creydr** [comment]
@cdalar thanks a lot for fixing this :+1: 
Lately some tests are flaky. You can simply retrigger a retest via the following:

/retest

**Pull request merged**